src/preparers.py

```diff
@@ -782,3 +932,30 @@ def prepare_bactrianx(row):
         {"from": "user", "text": inputs, "parent": row["_source"]},
         {"from": "assistant", "text": outputs, "parent": 0},
     ]
+
+
+def prepare_collective_cognition(row):
+    """
+    Prepares a dataset row by converting conversation parts into a standardized messages format.
+
+    Args:
+        row (dict): A single dataset row containing model information and conversations.
+
+    Returns:
+        list: A list of messages formatted as specified, with inputs and targets converted to messages.
+    """
+    messages = []
+    conversations = row.get("conversations", [])
+
+    if conversations:
+        for i, conversation in enumerate(conversations):
+            speaker = "user" if conversation["from"] == "human" else "assistant"
+            text = conversation["value"]
+            parent = "CC-chats-data-2023-10-16" if i == 0 else i - 1
+            if text is not None:
+                text = text.strip()
+            else:
+                text = ""
```

Comment:
Do we want to save messages with empty text?

Reply:
When the conversation ends it has empty text in some situations so I added this case.

Reply:
![Uploading image.png…]()

Reply:
![image](https://github.com/Data-Provenance-Initiative/Data-Provenance-Collection/assets/25876527/37789e79-c240-4500-ab61-8643dc24eaa4)